scripts/pre_commit_linter.py

```diff
@@ -1068,24 +1068,8 @@ def _lint_all_files(self):
             process.daemon = False
             process.start()
 
-        file_groups_to_lint = [
-            html_files_to_lint_for_css, css_files_to_lint,
-            js_files_to_lint, py_files_to_lint]
-        number_of_files_to_lint = sum(
-            len(file_group) for file_group in file_groups_to_lint)
-
-        timeout_multiplier = 2000
-        for file_group, process in zip(file_groups_to_lint, linting_processes):
-            # try..except block is needed to catch ZeroDivisionError
-            # when there are no CSS, HTML, JavaScript and Python files to lint.
-            try:
-                # Require timeout parameter to prevent against endless
-                # waiting for the linting function to return.
-                process.join(timeout=(
-                    timeout_multiplier * (
-                        len(file_group) / number_of_files_to_lint)))
-            except ZeroDivisionError:
-                break
+        for process in linting_processes:
+            process.join()
```

Comment:
Can you please test this locally by making multiple errors to files in various dirs, focussing more on Python and JS files? Also, try running the check without any files. Does it terminate? Please ensure that all the linting processes terminate in reasonable time both locally and on CircleCI. In general, timeout prevents waiting endlessly for the process to return.